src/options.rs

```diff
@@ -118,6 +122,26 @@ impl<'a> Options<'a> {
         }
     }
 
+    /// Change [`self.tab_width`]. The tab width is how wide
+    /// a tab character should be considered. By default, a tab
+    /// character to be consiered to have a width of 0.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use textwrap::{Options, wrap};
+    ///
+    /// let options = Options::new(16).tab_width(4);
+    /// assert_eq!(wrap("This\tis\tan\texample.", options),
+    ///            vec!["This\tis\tan",
```

Comment:
If I didn't know anything about the crate, I think this example would confuse me: it looks like it shows that we split text on `\t`. However, I believe what actually happens here is that

1. `tab_width` is taken into account — the new behavior, which is great! — and then
2. `break_words` makes us break the line on the `\t`

It's hard to tell because `"This\tis\tan"` happens to have 8 letters and 2 TABs, each of which is 4 columsn wide. So the string fits precisely in the width of 16 columns allowed.

We should figure out if we start counting TAB as a word separator. Until then, a better illustration of `tab_width` would be something like this:

```rust
assert_eq!(wrap("foo\tbar baz", Options::new(12).tab_width(0)), vec!["foo\tbar baz"]);
assert_eq!(wrap("foo\tbar baz", Options::new(12).tab_width(4)), vec!["foo\tbar", "baz"]);
```

Reply:
Fair point! It's also a very good thing you suggested this, because it revealed a bug I'd introduced in `wrap_single_line`. It looks like the condition that I've updated there in 046911e8530bc70cee13de1f3485f56a646bae62 makes the assumption that a string's display width is no longer than its byte length, which is no longer true since tabs only take one byte, but may be configured to take up much more width than that. Does the change I've made there seem like a reasonnable solution to you?

Reply:
Looks like `fill`'s fast path had the same issue. I fixed it the same way in e9d18ce8da6fa8379e6fc278ccbf56c35b61720b.

Reply:
> It looks like the condition that I've updated there in [046911e](https://github.com/mgeisler/textwrap/commit/046911e8530bc70cee13de1f3485f56a646bae62) makes the assumption that a string's display width is no longer than its byte length, which is no longer true since tabs only take one byte, but may be configured to take up much more width than that. Does the change I've made there seem like a reasonnable solution to you?

Oh, that is a good catch! I had not thought of that at all :sweat_smile: I think the fix looks good! You bring up a good point about the interaction of the fast path and the tab with in your other comment, so let me comment there as well.